table/column.go

```diff
@@ -88,16 +88,16 @@ func ToColumn(col *model.ColumnInfo) *Column {
 // If pkIsHandle is false and name is ExtraHandleName, the extra handle column will be added.
 // If any columns don't match, return nil and the first missing column's name
 func FindCols(cols []*Column, names []string, pkIsHandle bool) ([]*Column, string) {
-	var rcols []*Column
-	for _, name := range names {
+	var rcols = make([]*Column, 0, len(names))
+	for i, name := range names {
 		col := FindCol(cols, name)
 		if col != nil {
 			rcols = append(rcols, col)
 		} else if name == model.ExtraHandleName.L && !pkIsHandle {
 			col := &Column{}
 			col.ColumnInfo = model.NewExtraHandleColInfo()
 			col.ColumnInfo.Offset = len(cols)
-			rcols = append(rcols, col)
+			rcols[i] = col
```

Comment:
seem you still need to use append here.

Reply:
Already fixed.
And the func FindCol also a slow liner-find

Reply:
yes, I have already told @zz-jason about this, maybe we can use a map to speed up it.

Reply:
@winoros PTAL

Reply:
@siddontang https://github.com/pingcap/tidb/pull/14903 using a map